File: ackindex/ack.erl
%RUN IT BY
%   erlc ack.erl; time echo 'ack:start().'|erl

-module(ack).
-export([start/0]).

ack(0, _, Y) -> Y + 1;
ack(1, X, 0) -> X;
ack(2, _, 0) -> 0;
ack(_, _, 0) -> 1;
ack(N, X, Y) -> ack(N - 1, X, ack(N, X, Y - 1)).

start() -> ack(5, 2, 3).

File: ackindex/ack.c
#include <stdio.h>

long ack(long n, long x, long y) {
   if (n == 0) return y + 1;
   if (y == 0)
      switch (n) {
         case 1: return x;
         case 2: return 0;
         default: return 1;
      }
   return ack(n - 1, x, ack(n, x, y - 1));
}

main() {
   printf("%ld\n", ack(5, 2, 3));
}

